### PR TITLE
perf: optimize not to update partitions with newly created level 2 files

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Beside the optimization to the function the trigger invokes, this PR also remove non-longer used function

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: compactor/src/compact.rs [hunk 1]
```diff
@@ -927,23 +927,13 @@ pub mod tests {
         };
         let _pf2 = txn.parquet_files().create(p2).await.unwrap();
         txn.commit().await.unwrap();
-        // 1 candidate becasue we limit it
-        let candidates = compactor
-            .partitions_to_compact(compaction_type, vec![hour_threshold], 1)
-            .await
-            .unwrap();
-        assert_eq!(candidates.len(), 1);
-        // 2 candidates because we do not limit it
```
Comment (NGA-TRAN) [inline compactor/src/compact.rs:936]: Due to the optimization, there is only 1 partition returned here. The limit test is moved to the end of this test

File: iox_catalog/src/interface.rs [hunk 2]
```diff
@@ -634,15 +634,6 @@ pub trait ParquetFileRepo: Send + Sync {
...
 
-    // Remove this function: https://github.com/influxdata/influxdb_iox/issues/6518
-    /// Select partition for cold/warm/hot compaction
-    /// These are partitions with files created recently (aka created after the specified time_in_the_past)
-    /// These files include all levels of compaction files and both non-deleted and soft-deleted files
-    async fn partitions_with_recent_created_files(
-        &mut self,
-        time_in_the_past: Timestamp,
-    ) -> Result<Vec<PartitionId>>;
-
```
Comment (NGA-TRAN) [inline iox_catalog/src/interface.rs:645]: This function has not been used for a week

File: iox_catalog/src/mem.rs [hunk 3]
```diff
@@ -1195,15 +1195,19 @@ impl ParquetFileRepo for MemTxn {
...
-        partition.new_file_at = Some(created_at);
+        // Only update if the compaction level is not Final which signal more compaction needed
+        if compaction_level < CompactionLevel::Final {
+            let partition = stage
+                .partitions
+                .iter_mut()
+                .find(|p| p.id == partition_id)
+                .ok_or(Error::PartitionNotFound { id: partition_id })?;
+            partition.new_file_at = Some(created_at);
+        }
```
Comment (NGA-TRAN) [inline iox_catalog/src/mem.rs:1210]: for mem catalog, the trigger of creating a new file is implemented here. This change is just a corresponding trigger update

**Review by domodwyer — APPROVED** [review]
Trigger looks good - didn't check the tests (I assume they're fairly mechanical changes).

Thanks for cleaning up unused catalog functions 🙏

**Pull request merged**